Sources/Git/Branch.swift

```diff
@@ -1,15 +1,43 @@
 import Foundation
 
-public protocol BranchInteractor {
-	func getCurrentBranch() throws -> String
-	func getAllBranches(from source: Branch.Source) throws -> String
+protocol BranchInteractor {
+	func getCurrentBranch(atPath path: String) throws -> String
+	func getAllBranches(from source: Branch.Source, atPath path: String) throws -> String
 }
 
-public struct Branch: Equatable {
-	public enum Source: Equatable {
+public struct Branch: Equatable, Codable {
+	public enum Source: Equatable, RawRepresentable {
 		case local
 		case remote(Remote)
 
+		public init?(rawValue: String) {
+			let components = rawValue.components(separatedBy: .whitespaces)
+			let component1 = components.first
+			let component2 = components.dropFirst().first
+			let component3 = components.dropFirst(2).first
+
+			switch (component1, component2, component3) {
+			case ("local", _, _):
+				self = .local
+			case ("remote", let name?, let urlString?):
```

Comment:
<!--
  2 failure:  Combine multiple ..., Combine multiple ...
  0 warning: 
  
  
  DangerID: danger-id-default;
  File: Sources/Git/Branch.swift;
  Line: 22;
-->
- :no_entry_sign: Combine multiple pattern matching bindings by moving keywords out of tuples. (`pattern_matching_keywords`)
- :no_entry_sign: Combine multiple pattern matching bindings by moving keywords out of tuples. (`pattern_matching_keywords`)